### ISSUE TITLE
[Bug] [2020.04] TextInput onChange lose focus

### ISSUE THREAD

**Issue by aljesusg** [patternfly/patternfly-react#4072]
After upgrade when the value change the textInput lost focus, it seems that is rendering again the component in onChange event. (See Gif)

## Component 
```ts
<TextInput
                value={this.state.addEgressHost.host}
                type="text"
                id="addEgressHost"
                key="addEgressHost"
                aria-describedby="add egress host"
                name="addHost"
                onChange={this.onAddHost}
                isValid={this.state.validEgressHost}
              />
```

## OnAddHost method

```ts
  onAddHost = (value: string, _) => {
    const host = value.trim();
    this.setState({
      addEgressHost: {
        host: host
      },
      validEgressHost: isServerHostValid(host)
    });
  };
```

## GIF

![ezgif-4-5a5a26e8e121](https://user-images.githubusercontent.com/3019213/79236783-b4ea2400-7e6d-11ea-900a-84702a79ede2.gif)

**Comment by aljesusg** [comment]
@redallen Do you know who I can talk to about this? Thanks !!!

**Comment by redallen** [comment]
Taking a look now.

**Comment by redallen** [comment]
I'm unable to reproduce this using your code (here's the codesandbox I made that doesn't seem to have your problem: https://codesandbox.io/s/autumn-framework-w26vd). The only change we made to TextInput in our 2020.04 release was my PR #3919 for fixing forward ref types, are you by chance passing a ref?

If you can link a reproducible example or tell me how to load this problematic page while running kiali I can help further.

**Comment by aljesusg** [comment]
Ok @redallen let me do some research. Thanks !!!

**Comment by aljesusg** [comment]
@redallen After some tests I found the problem, this happens when the component is in a table. So something changed here.

Here you have the sandbox https://codesandbox.io/s/texinput-table-ggx1l?file=/index.js

Is similar to our code https://github.com/kiali/kiali-ui/blob/master/src/pages/IstioConfigNew/GatewayForm.tsx

Somethign change here maybe we shopuld cast this to ICell ?? something change when PF render this.

**Comment by redallen** [comment]
I'm investigating this now and it seems to be a problem on _any_ `<input>` element inside of a table row. It also appears to occur on `"@patternfly/react-table": "2.24.41"` which is what kiali is using. Will update once I find a fix...

**Comment by redallen** [comment]
Not quite sure the root cause yet, but a temporary fix is to use `TextInputBase` instead of `TextInput`. I'm debugging more of why our `Table` component doesn't like forwarded refs...

**Comment by aljesusg** [comment]
Ok thanks @redallen , I'm going to check TextInputBase. Thanks !!!

**Comment by redallen** [comment]
TLDR; Another table bug.

I've spent most of the day on this and am bewildered. My next step would be to make a minimal reproduction with a stripped down version of our `Table` component, but I'm not quite sure how to do that yet since I lack a good understanding of how Table works with React's lifecycle methods.

To sum up what I've found, using `TextInputBase` causes typing in the input to only update `TextInputBase` inside the `<Table>`. However, using `React.forwardRef` on `TextInputBase` causes the whole `<RowWrapper>` (and maybe even higher up the tree?) to rerender, and so the `<input>` loses focus. I assume this has to do [internally with how React implements forwardRef](https://github.com/facebook/react/pull/12346/files/5b5b625a055a86a96198babf90f47034497204f4) when diff-checking. However, the fault is in our Table since `<div>{this.rows()[0].cells}</div>` works just fine.

The key implementation takeaway for `TextInputBase` vs `TextInput` is that `TextInputBase` is a normal React Component (that is `typeof TextInputBase === 'function'`), while `TextInput` is a special React object (that is `typeof TextInput === 'object'`) with a `.render` prop.

[I've created a branch](https://github.com/redallen/patternfly-react/tree/fix/table-input-focus) for debugging if anyone wants to further investigate.

**Comment by aljesusg** [comment]
ok @redallen is working with TextInputBase we are moving forward with this and I'm going to document this to change it in next PF4 release or after fix. Thanks

**Comment by dgdavid** [comment]
Hi there!

Just for the record, is spite of bug report is closed this still reproducible when using

```json
    ...
   
    "@patternfly/patternfly": "4.50.4",
    "@patternfly/react-core": "4.63.3",
    "@patternfly/react-table": "^4.18.14",
   
    ...
```

Regards.
